acvm-repo/acvm/src/compiler/transformers/csat.rs

```diff
@@ -66,9 +66,11 @@ impl CSatTransformer {
         self.solvable_witness.insert(witness);
     }
 
-    // Still missing dead witness optimization.
-    // To do this, we will need the whole set of assert-zero opcodes
-    // I think it can also be done before the local optimization seen here, as dead variables will come from the user
+    /// Transform the input arithmetic expression into a new one having the correct 'width'
+    /// by creating intermediate variables as needed.
+    /// Having the correct width means:
+    /// - it has at most one multiplicative term
+    /// - it has at most 'width-1' linear combinations terms.
```

Comment:
Should we add it has at most `width` unique witnesses?

Reply:
I added an explanation of why it is width1 instead.